construct/core.py

```diff
@@ -524,10 +524,12 @@ def _parse(self, stream, context):
         except ConstructError:
             if c < self.mincount:
                 raise RangeError("expected %d to %d, found %d" %
-                    (self.mincount, self.maxcout, c), sys.exc_info()[1])
+                    (self.mincount, self.maxcout, c))
             stream.seek(pos)
         return obj
     def _build(self, obj, stream, context):
+        if not hasattr(obj, '__len__') or hasattr(obj, 'get'):
```

Comment:
operator prececence, not higher than or
https://docs.python.org/3/reference/expressions.html#operator-precedence